lib/kitchen/ancestor.rb

```diff
@@ -38,7 +38,11 @@ def increment_descendant_count(descendant_type)
     # @param by [Integer] the amount by which to decrement
     #
     def decrement_descendant_count(descendant_type, by: 1)
-      @descendant_counts[descendant_type.to_sym] = get_descendant_count(descendant_type) - by
+      if @descendant_counts.has_key?(descendant_type.to_sym)
+        @descendant_counts[descendant_type.to_sym] = get_descendant_count(descendant_type) - by
+      else
+        raise "An element cannot have negative descendants"
```

Comment:
Great idea to have this protection!  However note that it does not fully protect against negative counts.  If the count already existed at `1` and you had `by: -2` it'd go negative.  You could add that spec first and then fix here.

Reply:
Also modify the YARD documentation above this method that this method can raise an error (there are other examples of this)